src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -291,6 +290,21 @@ protected unsafe void WriteUTF8Chars(string value)
             }
         }
 
+        protected unsafe void WriteUTF8Bytes(ReadOnlySpan<byte> value)
+        {
+            if (value.Length < bufferLength)
+            {
+                byte[] buffer = GetBuffer(value.Length, out int offset);
+                value.CopyTo(new Span<byte>(buffer, offset, buffer.Length));
```

Comment:
```suggestion
                value.CopyTo(buffer.AsSpan(offset));
```
If the code in this PR is passing tests, it seems like we're missing some tests, as a non-0 offset value here will result in exceptions.